test/jdk/com/sun/net/httpserver/simpleserver/CustomFileSystemTest.java

```diff
@@ -422,19 +424,31 @@ public void testSymlinkSegmentGET() throws Exception {
         var symlink = root.resolve("symlink");
         var target = Files.createDirectory(root.resolve("target"));
         Files.writeString(target.resolve("aFile.txt"), "some text", CREATE);
-        Files.createSymbolicLink(symlink, target);
+        if (createSymLink(symlink, target)) {  // sym link creation succeeded
 
-        var server = SimpleFileServer.createFileServer(LOOPBACK_ADDR, root, OutputLevel.VERBOSE);
-        server.start();
+            var server = SimpleFileServer.createFileServer(LOOPBACK_ADDR, root, OutputLevel.VERBOSE);
+            server.start();
+            try {
+                var client = HttpClient.newBuilder().proxy(NO_PROXY).build();
+                var request = HttpRequest.newBuilder(uri(server, "symlink/aFile.txt")).build();
+                var response = client.send(request, BodyHandlers.ofString());
+                assertEquals(response.statusCode(), 404);
+                assertEquals(response.headers().firstValue("content-length").get(), expectedLength);
+                assertEquals(response.body(), expectedBody);
+            } finally {
+                server.stop(0);
+            }
+        }
+    }
+
+    private boolean createSymLink(Path symlink, Path target) {
```

Comment:
Seems like createSymLink should actually return true or false, or if it throws an exception in one case, then it should be void, in which case you wouldn't need the if statement in the calling code

Reply:
Good catch, that makes for cleaner code - I'll refactor.